### PR TITLE
gh-104012: Ensure test_calendar.CalendarTestCase.test_deprecation_warning consistently passes

### LINKED ISSUE #104012: `test_calendar` fails on Azure Pipelines
The newly added test `test_calendar.CalendarTestCase.test_deprecation_warning` (added in https://github.com/python/cpython/commit/84e7d0f0c7f9a44d81be2d705ed4d401a6505356) has been consistently failing on Azure Pipelines over the past day. Example: https://dev.azure.com/Python/cpython/_build/results?buildId=126290&view=logs&j=0fcf9c9b-89fc-526f-8708-363e467e119e&t=fa5ef4ee-3911-591e-4444-19482ab189b7&l=463. I'm not sure why there's a difference between Azure Pipelines and GitHub Actions here, but it does also fail for me locally on Windows.

Cc. @Agent-Hellboy / @hugovk

The fix seems simple; I have a PR ready.

### Linked PRs
* gh-104014

### PR DESCRIPTION
**Pull request by AlexWaygood**
Closes #104012

* Issue: gh-104012

### CODE REVIEW

File: Lib/test/test_calendar.py [hunk 1]
```diff
@@ -493,6 +493,7 @@ def test_format(self):
 class CalendarTestCase(unittest.TestCase):
 
     def test_deprecation_warning(self):
+        warnings.resetwarnings()
```
Comment (JelleZijlstra) [inline Lib/test/test_calendar.py]: Why aren't we using `self.assertWarns`?

Comment (AlexWaygood) [reply Lib/test/test_calendar.py]: I don't have a reason not to use `self.assertWarns`. I just made the minimal change to fix the issue. Looking through the previous PR, it doesn't look like using `assertWarns(Regex)` was suggested, so the contributor may not have been aware of it.

Comment (JelleZijlstra) [reply Lib/test/test_calendar.py]: Let's switch to `assertWarnsRegex`, then we don't have to do all this manual manipulation of the warnings machinery.

Comment (AlexWaygood) [reply Lib/test/test_calendar.py]: yeah, sounds good

**Review by JelleZijlstra — APPROVED** [review]

**Pull request merged**